src/v2/Artsy/Relay/middleware/cache/cacheMiddleware.ts

```diff
@@ -0,0 +1,93 @@
+import { QueryResponseCache } from "relay-runtime"
+import { isFunction } from "lodash"
+import createLogger from "v2/Utils/logger"
+import { Cache, CacheConfig } from "./Cache"
+
+const logger = createLogger("v2/Artsy/middleware/cache/cacheMiddleware")
+
+interface CacheMiddlewareOpts extends CacheConfig {
+  onInit?: (cache: QueryResponseCache) => any
+  allowMutations?: boolean
+  allowFormData?: boolean
+  clearOnMutation?: boolean
+  cacheErrors?: boolean
+  disableServerSideCache?: boolean
+}
+
+/**
+ * Adapted from `react-relay-network-modern`.
+ * @see https://github.com/relay-tools/react-relay-network-modern/tree/master/src/middlewares
+ */
+export function cacheMiddleware(opts?: CacheMiddlewareOpts) {
+  const {
+    size,
+    ttl,
+    onInit,
+    allowMutations,
+    allowFormData,
+    clearOnMutation,
+    cacheErrors,
+    disableServerSideCache = true,
+  } = opts || {}
+
+  const cache = new Cache({
+    size: size || 100, // 100 requests
+    ttl: ttl || 15 * 60 * 1000, // 15 minutes
+    disableServerSideCache,
+  })
+
+  if (isFunction(onInit)) {
+    onInit(cache)
+  }
+
+  return next => async req => {
+    if (req.isMutation()) {
+      if (clearOnMutation) {
+        await cache.clear()
```

Comment:
This appears to do a `flushAll` which would clear the _entire_ redis database. Is this intentional? (I may be misunderstanding.)

Reply:
Currently when we perform a mutation we completly clear out the relay cache. But thinking further, we don't have any unauthenticated mutations in force that we need concern ourselves with so this can go away. 

(Also note we're not really using redis in force ATM, so if this continues to scale up through our new application infrastructure it would ideal to keep things in sync with relay store, esp if we start to extend cache behavior to selective logged in requrests.)